### PR TITLE
Devel/abi/mbc mods

### PR DESCRIPTION
**Pull request by choobacca-bacca**
Have added Safety Integrated System and tested it locally, have also added Dockerfile template to be used for adding to Kubernetes Cluster

### CODE REVIEW

**Review by luca-della-vedova — COMMENTED** [review]
Looks reasonable from a high level view.
I think the main change I would like to see is make the timeouts ros parameters that are parsed from a yaml, then when this is deployed we can just mount the yaml as a config map in the kubernetes cluster and update the timeouts without having to rebuild the image.

File: trl_safety_system/src/LiftStateSubscriber.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  if ((lift_mode == rmf_lift_msgs::msg::LiftState::MODE_AGV) &&
+    (lift_door_state != rmf_lift_msgs::msg::LiftState::DOOR_CLOSED))
+  {
+    RCLCPP_INFO(node_handle_->get_logger(), "Lift door is open in AGV mode");
+    rclcpp::Duration time_diff = node_handle_->get_clock()->now() - last_known_time_;
+    if (time_diff > rclcpp::Duration(timeout_duration_, 0.0) && !notification_triggered_) {
+      Aws::String notification_message =
+        "Lift has an issue, It has been stuck at the same floor in AGV mode with doors open";
+      alert_notifier_pointer_->publishSNSMessage(notification_message);
+      notification_triggered_ = true;
```
[the remark this answers is not in the document]
Comment (choobacca-bacca) [reply trl_safety_system/src/LiftStateSubscriber.cpp:38]: Ok have reset this variable in the same code block as `last_known_time_` as recommended

File: trl_safety_system/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# include("/opt/vcpkg/scripts/buildsystems/vcpkg.cmake")
+
+set(BUILD_SHARED_LIBS ON CACHE STRING "/usr/local/lib")
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(AWSSDK REQUIRED COMPONENTS s3 sns)
```
[the remark this answers is not in the document]
Comment (choobacca-bacca) [reply trl_safety_system/CMakeLists.txt]: have added a warning message to be published if AWSSDK not found and where to install it from

**Comment by choobacca-bacca** [comment]
Ok have made the adjustments as well as have added the config file, more changes to be done, but I believe this is good to go for first merge.

**Review by luca-della-vedova — APPROVED** [review]

**Pull request merged**